src/script/components/collection/Collection.tsx

```diff
@@ -0,0 +1,225 @@
+      if (message.isExpired()) {
+        return categories;
+      }
+
+      if (isOfCategory('images', message)) {
+        categories.images.push(message);
+      } else if (isOfCategory('audio', message)) {
+        categories.audio.push(message);
+      } else if (isOfCategory('files', message)) {
+        categories.files.push(message);
+      } else if (isOfCategory('links', message)) {
+        categories.links.push(message);
+      }
+      return categories;
+    },
+    {
+      audio: [],
+      files: [],
+      images: [],
+      links: [],
+    },
+  );
+}
+
+const Collection: React.FC<CollectionDetailsProps> = ({conversation, conversationRepository}) => {
+  const [searchTerm, setSearchTerm] = useState('');
+  const {display_name} = useKoSubscribableChildren(conversation, ['display_name']);
+  const [messages, setMessages] = useState<ContentMessage[]>([]);
+  const [detailCategory, setDetailCategory] = useState<Category | undefined>(undefined);
+
+  useEffect(() => {
+    conversationRepository
+      .getEventsForCategory(conversation, MessageCategory.LINK_PREVIEW)
+      .then(allMessages => setMessages(allMessages as ContentMessage[]));
+  }, []);
+
+  useEffect(() => {
+    const addItem = (message: ContentMessage) => {
+      setMessages(oldMessages => [message].concat(oldMessages));
+    };
+
+    const removeItem = (messageId: string, conversationId: string) => {
+      if (conversation.id !== conversationId) {
+        // A message from a different converation, nothing to do
+        return;
+      }
+      setMessages(oldMessages => oldMessages.filter(message => message.id !== messageId));
+    };
+
+    const removeMessage = (message: ContentMessage) => {
+      removeItem(message.id, message.conversation_id);
+    };
+
+    amplify.subscribe(WebAppEvents.CONVERSATION.EPHEMERAL_MESSAGE_TIMEOUT, removeMessage);
+    amplify.subscribe(WebAppEvents.CONVERSATION.MESSAGE.ADDED, addItem);
+    amplify.subscribe(WebAppEvents.CONVERSATION.MESSAGE.REMOVED, removeItem);
+    return () => {
+      amplify.unsubscribe(WebAppEvents.CONVERSATION.EPHEMERAL_MESSAGE_TIMEOUT, removeMessage);
+      amplify.unsubscribe(WebAppEvents.CONVERSATION.MESSAGE.ADDED, addItem);
+      amplify.unsubscribe(WebAppEvents.CONVERSATION.MESSAGE.REMOVED, removeItem);
+    };
+  }, []);
+
+  const categories = splitIntoCategories(messages);
+  const {images, audio, links, files} = categories;
+
+  if (detailCategory && categories[detailCategory].length > 0) {
+    return (
+      <CollectionDetails
+        conversation={conversation}
+        messages={categories[detailCategory]}
+        onClose={() => setDetailCategory(undefined)}
+      />
+    );
+  }
+
+  const content = searchTerm ? null : (
+    <>
+      <Section
+        messages={images}
+        limit={12}
+        uieName={'collection-section-image'}
+        onSelect={() => setDetailCategory('images')}
+      >
+        <span className={`collection-header-icon icon-library`}></span>
```

Comment:
maybe this could be also included and just be a conditional for this or the MicOn icon?

Reply:
depends on how reusable we want this component to be.